File: vue_type_script/src/main/java/com/lasgis/prototype/vue/controller/UserController.java
/*
 *  @(#)UserController.java  last: 13.06.2023
 *
 * Title: LG prototype for java-spring-jdbc + vue-type-script
 * Description: Program for support Prototype.
 * Copyright (c) 2023, LasGIS Company. All Rights Reserved.
 */

package com.lasgis.prototype.vue.controller;

import com.lasgis.prototype.vue.exception.UserNotFoundException;
import com.lasgis.prototype.vue.model.UserDto;
import com.lasgis.prototype.vue.model.UserRole;
import com.lasgis.prototype.vue.repository.UserDao;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Random;

@Slf4j
@RestController
@RequestMapping("/api")
public class UserController {

    public static final String HELLO_TEXT = "Hello from Spring Boot Backend!";
    public static final String SECURED_TEXT = "Hello from the secured resource!";
    public static final Random RANDOM_ID = new Random();
    private final UserDao userDao;

    public UserController(UserDao userDao) {
        this.userDao = userDao;
    }

    @ResponseBody
    @GetMapping(path = "/hello")
    public String sayHello() {
        log.info("GET called on /hello resource");
        return HELLO_TEXT;
    }

    @ResponseBody
    @PostMapping(path = "/user/{name}/{login}")
    @ResponseStatus(HttpStatus.CREATED)
    public long addNewUser(@PathVariable("name") String name, @PathVariable("login") String login) {
        UserDto newUser = UserDto.builder().userId(RANDOM_ID.nextInt())
            .login(login)
            .name(name)
            .roles(List.of(UserRole.OPERATOR))
            .build();
        Integer userId = userDao.insert(newUser);
        newUser.setUserId(userId);
        log.info(newUser + " successfully saved into DB");
        return userId;
    }

    @ResponseBody
    @GetMapping(path = "/user/{id}")
    public UserDto getUserById(@PathVariable("id") int id) {
        return userDao.findById(id).map(user -> {
            log.info("Reading user with id " + id + " from database.");
            return user;
        }).orElseThrow(() -> {
            throw new UserNotFoundException("The user with the id " + id + " couldn't be found in the database.");
        });
    }

    @ResponseBody
    @GetMapping(path = "/secured")
    public String getSecured() {
        log.info("GET successfully called on /secured resource");
        return SECURED_TEXT;
    }
}

File: demo.web/src/main/kotlin/com/lasgis/kotlin/demo/web/rest/UserController.kt
package com.lasgis.kotlin.demo.web.rest

import com.lasgis.kotlin.demo.web.dao.Greeting
import com.lasgis.kotlin.demo.web.mybatis.UserMapper
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.util.concurrent.atomic.AtomicLong

/**
 * <description>
 *
 * @author VLaskin
 * @since <pre>26.12.2019</pre>
 */
@RestController
class UserController(private val userMapper: UserMapper) {

    val counter = AtomicLong()

    @GetMapping("/greeting")
    fun greeting(@RequestParam(value = "name", defaultValue = "World") name: String) =
        Greeting(counter.incrementAndGet(), "Hello, $name")

    @GetMapping("/user")
    fun user(@RequestParam(value = "login", defaultValue = "VPupkin") login: String) =
        userMapper.findByLogin(login)
}